### PR TITLE
[ColorUtil] Extend Colour Temperature range 1000 K .. 10000 K

### PR DESCRIPTION
**Pull request by andrewfg**
Extend the supported colour temperature range of the `kelvinToXy()` method from 2000 .. 6500 K to 1000 .. 10000 K

See https://github.com/openhab/openhab-core/issues/3891#issuecomment-2428965622

Please note that the inverse `xyToKelvin()` method (which is based on McCamy's approximation) is only accurate above 2000 K. This means that round trip conversions `xyToKelvin(kelvinToXy())` are very inaccurate for colour temperatures below 2000 K. This is not perfect but it seems that most real bindings in OH are using a range of 2000 K to 6500 K .. where the accuracy is better than 1%..

Signed-off-by: AndrewFG <software@whitebear.ch>

### CODE REVIEW

**Comment by andrewfg** [comment]
=> @lolodomo for info

**Review by holgerfriedrich — COMMENTED** [review]
Hi @andrewfg, 
could we consider restricting the number of tests runs to a reasonable number? 270000 tests take quite a while to complete.
Mybe increasing ```allHSB::step``` to 20 and ```allRGBW::step``` to 25 could be a compromise.... WDYT?

**Comment by andrewfg** [comment]
Yeah. :)

File: bundles/org.openhab.core/src/test/java/org/openhab/core/util/ColorUtilTest.java [hunk 1]
```diff
@@ -435,7 +435,7 @@ private static Stream<Arguments> allXY() {
      */
     private static Stream<Arguments> allRGBW() {
         List<Arguments> result = new ArrayList<>();
-        final double step = 5.0;
+        final double step = 17.0;
```
Comment (holgerfriedrich) [inline bundles/org.openhab.core/src/test/java/org/openhab/core/util/ColorUtilTest.java:438]: Do you want to hit the upper boundary 100? Then a factor of 100 would be a better choice.
With step size 17, the loop will end at 85.

File: bundles/org.openhab.core/src/test/java/org/openhab/core/util/ColorUtilTest.java [hunk 2]
```diff
@@ -389,7 +389,7 @@ public Stream<? extends Arguments> provideArguments(@Nullable ExtensionContext c
      */
     private static Stream<Arguments> allHSB() {
         List<Arguments> result = new ArrayList<>();
-        final double step = 5.0;
+        final double step = 17.0;
```
Comment (holgerfriedrich) [inline bundles/org.openhab.core/src/test/java/org/openhab/core/util/ColorUtilTest.java:392]: Do you want to hit the upper boundaries 100 and 360? Then a factor of 100 and 360 would be a better choice.
With step size 17, the loop will end at 85 and 357, respectively. 10 or 20 might be a better choice.

File: bundles/org.openhab.core/src/test/java/org/openhab/core/util/ColorUtilTest.java (hunk 1 above)
Comment (andrewfg) [reply bundles/org.openhab.core/src/test/java/org/openhab/core/util/ColorUtilTest.java:438]: I know. I was trying to set a step that tests intersticial values between the obvious round numbers. So I was thinking of using primes. But perhaps a factor of 100 would indeed be better..

File: bundles/org.openhab.core/src/test/java/org/openhab/core/util/ColorUtilTest.java (hunk 2 above)
Comment (andrewfg) [reply bundles/org.openhab.core/src/test/java/org/openhab/core/util/ColorUtilTest.java:392]: See above.

**Review by holgerfriedrich — APPROVED** [review]
LGTM, thanks!

**Pull request merged**